From 3f9907f93d3983033e176e95f5134a57900a7f6e Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@asymptotic.io>
Date: Thu, 10 Dec 2020 19:06:31 -0500
Subject: [PATCH] build: Use cmake to look up abseil dependency

This should be much more robust than looking up the library directly.

Fixes: https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/issues/4
---
 meson.build    | 21 ++++++++-------------
 2 files changed, 10 insertions(+), 14 deletions(-)

diff --git a/meson.build b/meson.build
index bae155e..96d84d7 100644
--- a/meson.build
+++ b/meson.build
@@ -40,19 +40,14 @@ os_deps = []
 have_posix = false
 have_win = false
 
-absl_dep = [
-  cpp.find_library('absl_base'),
-  cpp.find_library('absl_bad_optional_access'),
-  cpp.find_library('absl_flags_internal'),
-  cpp.find_library('absl_flags_marshalling'),
-  cpp.find_library('absl_flags_parse'),
-  cpp.find_library('absl_flags_registry'),
-  cpp.find_library('absl_flags_usage_internal'),
-  cpp.find_library('absl_raw_logging_internal'),
-  cpp.find_library('absl_strings'),
-  cpp.find_library('absl_synchronization'),
-  cpp.find_library('absl_throw_delegate'),
-]
+absl_dep = dependency('absl', method : 'cmake',
+  modules : [
+    'absl::base',
+    'absl::flags_parse',
+    'absl::strings',
+    'absl::synchronization',
+  ]
+)
 
 if ['darwin', 'ios'].contains(host_system)
   os_cflags = ['-DWEBRTC_MAC']
-- 
GitLab

